my_compassion/models/res_user.py

```diff
@@ -31,3 +34,176 @@ def reset_password(self, login):
+            selected_partner.signup_url = old_partner.signup_url
+            old_partner.unlink()
+            _logger.info("Partner deleted.")
+
+            _logger.info(
+                "User %s linked to existing partner %s asynchronously.",
+                user.login,
+                selected_partner.name,
+            )
+        else:
+            # No better partner found
+            _logger.info("No better partner found for user %s.", user.login)
+
+        self.flush()
+
+    @api.model
+    def signup(self, values, token=None):
+        """Override signup to avoid creating unnecessary partners."""
+        values.pop("privacy_policy", None)
+
+        if token:
+            # Existing logic for token-based signup
+            partner = self.env["res.partner"]._signup_retrieve_partner(
+                token, check_validity=True, raise_exception=True
+            )
+            partner.write(
+                {
+                    "signup_token": False,
+                    "signup_type": False,
+                    "signup_expiration": False,
+                }
+            )
+
+            partner_user = partner.user_ids and partner.user_ids[0] or False
+
+            if partner_user:
+                values.pop("login", None)
+                values.pop("name", None)
+                partner_user.write(values)
+                if not partner_user.login_date:
+                    partner_user._notify_inviter()
+                return (self.env.cr.dbname, partner_user.login, values.get("password"))
+            else:
+                values.update(
+                    {
+                        "name": partner.name,
+                        "partner_id": partner.id,
+                        "email": values.get("email") or values.get("login"),
+                    }
+                )
+                if partner.company_id:
+                    values["company_id"] = partner.company_id.id
+                    values["company_ids"] = [(6, 0, [partner.company_id.id])]
+                partner_user = self._signup_create_user(values)
+                partner_user._notify_inviter()
+        else:
+            # No token: create a user with a temporary partner
+            email = values.get("email") or values.get("login")
+
+            # Create the user and a partner.
+            new_user = self._signup_create_user(values)
+
+            # Ensure changes are flushed and committed before scheduling the job
+            self.flush()
+
+            # Queue the task to find or replace the partner
+            self.with_delay()._find_and_replace_partner_async(new_user.id, email)
+            _logger.info(
+                "Async task to find and replace partner for user %s queued.",
+                new_user.login,
+            )
+
+        return (self.env.cr.dbname, values.get("login"), values.get("password"))
+
+    def action_reset_password(self):
+        """create signup token for each user, and send their signup url by email"""
```

Comment:
I don't understand why we need to redefine the reset password action? Isn't the parent process working as needed?